students/aravichander/lesson08/circle_class.py

```diff
@@ -0,0 +1,54 @@
+from math import pi 
+circlelist = []
+
+#Creating class to represent circle object 
+class Circle:
+	def __init__(self,radius):
+		if radius == 0:
+			raise ValueError
+		elif radius < 0:
+			raise ValueError
+		else:
+			self.radius = radius
+			print("Circle with radius {0:.6f}".format(radius))
+	
+	@property
+	def diameter(self):
+		return self.radius*2
+
+	@diameter.setter
+	def diameter(self,diameter):
+		self.radius = (diameter)/2
+
+	@property
+	def area(self):
+		return pi*(self.radius**2)
+		#print("Area of circle is {0:.2f}".format(area))
+
+	#not fully understanding this but hoping I did this correctly!
+	@classmethod
+	def circle_from_diameter(cls, diameter):
+		radius = diameter/2
+		return cls.Circle(radius)
```

Comment:
not quite, but close -- in a classmethod, the first parameter (cls) is the class object itself -- in this case "Circle", so you can call it to make a new one:

``return cls(radius)``

make sure to write a test for this -- it won't run like it is!